mailbox/api/src/test/java/org/apache/james/mailbox/events/EventDeadLettersHealthCheckContract.java

```diff
@@ -0,0 +1,70 @@
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.mailbox.events;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.apache.james.core.Username;
+import org.apache.james.mailbox.MailboxSession;
+import org.apache.james.mailbox.model.MailboxConstants;
+import org.apache.james.mailbox.model.MailboxPath;
+import org.apache.james.mailbox.model.TestId;
+import org.junit.jupiter.api.Test;
+
+interface EventDeadLettersHealthCheckContract {
+
+    Username USERNAME = Username.of("user");
+    MailboxPath MAILBOX_PATH = new MailboxPath(MailboxConstants.USER_NAMESPACE, USERNAME, "mailboxName");
+    MailboxSession.SessionId SESSION_ID = MailboxSession.SessionId.of(235);
+    TestId MAILBOX_ID = TestId.of(563);
+    Event.EventId EVENT_ID_1 = Event.EventId.of("6e0dd59d-660e-4d9b-b22f-0354479f47b4");
+    Event.EventId EVENT_ID_2 = Event.EventId.of("6e0dd59d-660e-4d9b-b22f-0354479f47b5");
+    MailboxListener.MailboxAdded EVENT_1 = new MailboxListener.MailboxAdded(SESSION_ID, USERNAME, MAILBOX_PATH, MAILBOX_ID, EVENT_ID_1);
+    MailboxListener.MailboxAdded EVENT_2 = new MailboxListener.MailboxAdded(SESSION_ID, USERNAME, MAILBOX_PATH, MAILBOX_ID, EVENT_ID_2);
+    EventDeadLetters.InsertionId INSERTION_ID_1 = EventDeadLetters.InsertionId.of("6e0dd59d-660e-4d9b-b22f-0354479f47b7");
+    EventDeadLetters.InsertionId INSERTION_ID_2 = EventDeadLetters.InsertionId.of("6e0dd59d-660e-4d9b-b22f-0354479f47b8");
+
+    Group GROUP_A = new EventBusTestFixture.GroupA();
+    Group GROUP_B = new EventBusTestFixture.GroupB();
+
+    EventDeadLettersHealthCheck testee();
+
+    EventDeadLetters eventDeadLetters();
+
+    @Test
+    default void checkShouldReturnHealthyWhenEventDeadLetterHasNoEvent() {
+        assertThat(testee().check().isHealthy()).isTrue();
+    }
+
+    @Test
+    default void checkShouldReturnDegradedWhenEventDeadLetterHasEvent() {
+        eventDeadLetters().store(GROUP_A, EVENT_1, INSERTION_ID_1).block();
+
+        assertThat(testee().check().isDegraded()).isTrue();
+    }
+
+    @Test
+    default void checkShouldReturnDegradedWhenEventDeadLetterHasEvents() {
+        eventDeadLetters().store(GROUP_A, EVENT_1, INSERTION_ID_1).block();
+        eventDeadLetters().store(GROUP_B, EVENT_2, INSERTION_ID_2).block();
+
+        assertThat(testee().check().isDegraded()).isTrue();
```

Comment:
Would be better here if we can assert the degraded messages. You can use 
```java
assertThat(testee().check())
    .isEqualTo(new Degraded("a message...."));
```

Reply:
Yes, we can add more about it.